### DISCUSSION TITLE
Choppy ROR when using Imac Mini

### DISCUSSION THREAD

**Discussion by bolza4** [artisan-roaster-scope/artisan#1015] [Q&A] [answered]
Hello, 

the roastery I work for owns a Giesen w6 and I've been using Artisan on my windows PC without any issue, works like a charm.
however they bought an iMac Mini and I can't get a good setting for the ROR curve, looks very choppy as shown in the picture attached, while on the PC it looks very good. 

![IMG_3404](https://user-images.githubusercontent.com/108266430/203792460-4d0ee961-469b-4439-9e98-4b55a3b500ea.jpg)

both of them run the latest artisan version. 

PC sampling is 3s, iMac sampling is 4s because it can't handle a faster sampling.

PC smoothing settings are the following, and I get great results with them:
sampling 3s
smooth curves 0
Delta span 6s
Delta BT smoothing 22
no polyfit comp.

on the Mac I've tried:
smooth curves 5: curves are smooth, but too much time shift
Delta span 8s, 12s : no real improvement
Delta BT smoothing 32, 42: no real improvement, too much time shift

please help me out, I've spent 6 hours on a Sunday trying to fix it but I didn't get any good results.

thank you!

**Comment by bolza4** [comment]
Also, while on the PC I can get the LCD to show me the BT decimal numbers (like 198.3 °), on the iMac I've found no way to do that and I only get, for example, 156.0 °, 189.0°, without any decimal, which is important to me.

**Comment by MAKOMO** [comment, marked as answer]
Your setup is wrong. There are several setups for Giesen roasters as Giesen decided to charge extra for having decimals (only for ET/BT) send over its computer interface.

Without decimals the RoR computation suffers from what is called [Digital Noise](https://artisan-roasterscope.blogspot.com/2017/06/digital-noise.html).

This is known as "Cropster code" which needs to be purchased from Giesen and entered into the machine to unlock an extra interface which holds ET/BT with one decimal (yes, just one not 2 or 3 as you get from Probat for free).

Artisan comes with several machine setups for Giesen roasters as described on the [Artisan home page](https://artisan-scope.org/machines/giesen/). The WxA setups work without "Cropster code" but don't deliver any decimals while the WxA+ setups offer one decimal (only) for ET/BT. There are also setups supporting additional IR or ENV sensors for both variants.
